service/src/main/java/com/epam/esm/service/validator/impl/GiftCertificateValidator.java

```diff
@@ -0,0 +1,76 @@
+package com.epam.esm.service.validator.impl;
+
+import com.epam.esm.dto.GiftCertificateDto;
+import com.epam.esm.dto.TagDto;
+import com.epam.esm.service.validator.Validator;
+import org.springframework.stereotype.Component;
+
+import java.util.Set;
+import java.util.function.Predicate;
+
+/**
+ * Implemented {@link Validator} for {@link GiftCertificateDto}
+ */
+@Component
+public class GiftCertificateValidator implements Validator<GiftCertificateDto> {
+    private static final String GIFT_CERTIFICATE_NAME_REGEX_PATTERN = "^([A-Za-z ]{1,45})$";
```

Comment:
Propose to move duplicates regex to application configuration and inject them whenever it needed